### ISSUE TITLE
"ValueError: zero length field name in format" after upgrading to ansible-cmdb v1.25

### ISSUE THREAD

**Issue by peterny** [fboender/ansible-cmdb#151]
After upgrading to ansible-cmdb v1.25 I get this error:

```
[pnr-admin@linux-drift-01 tmp]$ ansible-cmdb -i ./tmp-hosts /kb/ansible-cmdb2/setup-dump/
Traceback (most recent call last):
  File "/usr/bin/ansible-cmdb", line 153, in <module>
    ansible = ansiblecmdb.Ansible(args, hosts_files, options.fact_cache, debug=options.debug)
  File "/usr/lib/ansible-cmdb/ansiblecmdb/ansible.py", line 61, in __init__
    self._handle_inventory(inventory_path)
  File "/usr/lib/ansible-cmdb/ansiblecmdb/ansible.py", line 82, in _handle_inventory
    self.log.debug("Determining type of inventory_path {}".format(inventory_path))
ValueError: zero length field name in format
[pnr-admin@linux-drift-01 tmp]$

```
In this case even with this totally simple, one-line inventory file:
```
[pnr-admin@linux-drift-01 tmp]$ cat ./tmp-hosts
pnr-test-01.kb.dk
[pnr-admin@linux-drift-01 tmp]$

```
My ansible version (upgraded yesterday) is:
```
[pnr-admin@linux-drift-01 tmp]$ ansible --version
ansible 2.4.2.0

```
Turning on debug doesn't reveal to me what is wrong (except it seems to happen during "Reading host facts"):

```
[pnr-admin@linux-drift-01 tmp]$ ansible-cmdb -d -i ./tmp-hosts /kb/ansible-cmdb2/setup-dump/
data_dir = /usr/lib/ansible-cmdb/ansiblecmdb/data
tpl_dir = /usr/lib/ansible-cmdb/ansiblecmdb/data/tpl
static_dir = /usr/lib/ansible-cmdb/ansiblecmdb/data/static
inventory files = ['./tmp-hosts']
template params = {'exclude_columns': None, 'data_dir': '/usr/lib/ansible-cmdb/ansiblecmdb/data', 'log': <logging.RootLogger instance at 0x7f74c5ec71b8>, 'lib_dir': '/usr/lib/ansible-cmdb/ansiblecmdb/data', 'version': '1.25', 'columns': None}
Parsing fact dir: /kb/ansible-cmdb2/setup-dump/
Reading host facts from /kb/ansible-cmdb2/setup-dump/pnr-test-01.kb.dk
Traceback (most recent call last):
  File "/usr/bin/ansible-cmdb", line 153, in <module>
    ansible = ansiblecmdb.Ansible(args, hosts_files, options.fact_cache, debug=options.debug)
  File "/usr/lib/ansible-cmdb/ansiblecmdb/ansible.py", line 61, in __init__
    self._handle_inventory(inventory_path)
  File "/usr/lib/ansible-cmdb/ansiblecmdb/ansible.py", line 82, in _handle_inventory
    self.log.debug("Determining type of inventory_path {}".format(inventory_path))
ValueError: zero length field name in format
[pnr-admin@linux-drift-01 tmp]$
```

Is ansible-cmdb v1.25 supposed to work with ansible 2.4.2.0 ?
Perhaps the host facts (ansible -m setup ...) format has changed?

My old ansible-cmdb v1.21 still (mostly) works with ansible 2.4.2.0 (although the group info is suddenly gone from the overview!?).

/ Peter

**Comment by fboender** [comment]
Are you using Python v2.6? Ansible-cmdb only supports Python 2.7 / 3.0 or higher. If ansible-cmdb v1.21 still works with python 2.6, that's more of a miracle / coincidence then a feature ;-) It might also explain why the group info is gone.

**Comment by peterny** [comment]
Yes, I was using python 2.6 (standard python version on my RHEL6 ansible server).
So I installed python 3.6 (newest available) via [Red Hat Software Collections](https://access.redhat.com/documentation/en-us/red_hat_software_collections/), i.e.:
```
sudo yum install rh-python36
```

After setting environment:
```
[pnr-admin@linux-drift-01 ~]$ source /opt/rh/rh-python36/enable
[pnr-admin@linux-drift-01 ~]$ which python
/opt/rh/rh-python36/root/usr/bin/python
[pnr-admin@linux-drift-01 ~]$ python --version
Python 3.6.3
```

... my ansible-cmdb does not seem to work at all:
```
[pnr-admin@linux-drift-01 ~]$ sudo yum list ansible-cmdb\*
Loaded plugins: product-id, search-disabled-repos, security, subscription-
              : manager
Installed Packages
ansible-cmdb.noarch             1.25-2              @/ansible-cmdb-1.25-2.noarch
[pnr-admin@linux-drift-01 ~]$ which ansible-cmdb
/usr/bin/ansible-cmdb
[pnr-admin@linux-drift-01 ~]$ ansible-cmdb --version
Traceback (most recent call last):
  File "/usr/lib/ansible-cmdb/ansiblecmdb/parser.py", line 12, in <module>
    import yaml
  File "/usr/lib/ansible-cmdb/yaml/__init__.py", line 2, in <module>
    from error import *
ModuleNotFoundError: No module named 'error'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/ansible-cmdb", line 19, in <module>
    import ansiblecmdb
  File "/usr/lib/ansible-cmdb/ansiblecmdb/__init__.py", line 1, in <module>
    from .parser import HostsParser, DynInvParser
  File "/usr/lib/ansible-cmdb/ansiblecmdb/parser.py", line 14, in <module>
    import yaml3 as yaml
ModuleNotFoundError: No module named 'yaml3'
[pnr-admin@linux-drift-01 ~]$
```
Is this a Python issue or an ansible-cmdb issue?

**Comment by peterny** [comment]
Additional info:
I now tried with 
`sudo yum install python27`
(from Red Hat Sotware Collections)
and after 
`source /opt/rh/python27/enable`
my ansible-cmdb v1.25 seems to work just fine (including group info).

I would still like your comment on the Python 3.6 issue above, though :-)

**Comment by fboender** [comment]
Oh, nice, you may have reproduced a bug that was informally reported to me which I couldn't reproduce myself. It seems to be a problem with the yaml module which uses things that have been deprecated somewhere between Python 3.4 and 3.6. I'll take a look at it.

**Comment by peterny** [comment]
Great!
I have a small (completely different) issue after my upgrades to ansible 2.4.2.0 and ansible-cmdb v1.25:
With the old versions unreachable servers was shown in the Host Overview with a red link  (e.g. "otrs-test-02.kb.dk", pointing to a detail section with  info like "No host information collected. Failed to connect to the host via ssh: ssh: connect to host otrs-test-02.kb.dk port 22: Connection timed out").
Now this unreachable server is just shown as plain text (no link, at the bottom of the overview).
Not a big issue, but I liked having unreachable servers highlighted (with red, and link to the ansible error) in the overview.

**Comment by fboender** [comment]
Would you mind creating a new issue for that? It was a feature, so if it broke, it should be fixed.

**Comment by fboender** [comment]
> Yes, I was using python 2.6 (standard python version on my RHEL6 ansible server). Stupid me!

FYI, I've added a warning when starting ansible-cmdb that alerts the user that Python v2.6 is not supported. You're not the first to run into this problem (and probably won't be the last). RHEL6 is still in common use, and as such so is Python 2.6.

**Comment by fboender** [comment]
The Python v3.6 issues has been fixed in the master branch, and will be included in the soon-to-be-released v1.26 of Ansible-cmdb.

The problem was that the yaml3 package wasn't being included in the packages or when doing 'make install'. So actually, ansible-cmdb shouldn't have worked with any Python v3.x version, unless running directly from the repository.

I hope to have ansible-cmdb v1.26 out today.
